### PR TITLE
added topic list and seperate topic pages

### PR DESCRIPTION
**Pull request by booayl**
added navigation bar for topics, user can now view articles filtered by topics

### CODE REVIEW

File: The-Puffin-Room/src/components/ArticleByTopics.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      })
+      .catch((error) => {
+        console.error("Error fetching articles:", error);
+        setIsLoading(false);
+      });
+  }, [page,topic]);
+
+  useEffect(() => {
+    getArticles();
+  }, [getArticles]);
```
Comment (nc-rose-mullan) [inline The-Puffin-Room/src/components/ArticleByTopics.jsx:32]: No need for getArticles in the dependency array here.

File: The-Puffin-Room/src/components/ArticleByTopics.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  const prevPage = () => {
+    if (page > 1) setPage(page - 1);
+  };
+
+  return (
+    <>
+      <h2>{reformattedTopic}</h2>
+      {isLoading ? (
+        <Loading isLoading={isLoading} />
```
Comment (nc-rose-mullan) [inline The-Puffin-Room/src/components/ArticleByTopics.jsx:46]: Surely if we make it down to line 46 isLoading will always be true - so we don't need to pass it into the component?

File: The-Puffin-Room/src/components/ArticleByTopics.jsx (hunk 1 above)
Comment (nc-rose-mullan) [reply The-Puffin-Room/src/components/ArticleByTopics.jsx:32]: This is really similar to your allArticles component, making it a bit non-DRY. How could you re-use your articles component to be able to deliver either all articles or just articles by a topic?

**Review by nc-rose-mullan — COMMENTED** [review]
Stylistically I really like this Angela! There are a couple of points of feedback to addres, though.

**Comment by booayl** [comment]
Hi Rose, thanks for the feedback! I actually made some big changes on the way I fetch topic when I am working on the sort articles ticket. So I am fetching topic using params now, and its all done in the  articles component :) 
and for some reason, it wouldn't work without the [getArticles] dependency 😅. And I have removed the loading 🥳

**Pull request merged**